This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Thu, 13 Feb 2020 16:00:00 +0100
Subject: [PATCH 1/3] Win32: ignore an error when loading the default config
 file

Windows simply does not have a default `fonts.conf` file and we
cannot assume that the `FONTCONFIG_FILE` or `FONTCONFIG_PATH`
environment variables are set. The default fallback configuration
file loaded from memory is usually sufficient.

diff --git a/src/fcxml.c b/src/fcxml.c
index 1111111..2222222 100644
--- a/src/fcxml.c
+++ b/src/fcxml.c
@@ -3346,8 +3346,10 @@ bail0:
     if (!ret && complain_again) {
 	if (name)
 	    FcConfigMessage (0, FcSevereError, "Cannot %s config file \"%s\": %s", load ? "load" : "scan", name, FcStrBufDoneStatic (&reason));
+#ifndef _WIN32
 	else
 	    FcConfigMessage (0, FcSevereError, "Cannot %s default config file: %s", load ? "load" : "scan", FcStrBufDoneStatic (&reason));
+#endif
 	FcStrBufDestroy (&reason);
 	return FcFalse;
     }

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Mon, 21 Jul 2025 10:43:07 +0200
Subject: [PATCH 2/3] Silence FcInit warnings written to stderr

See:
https://gitlab.freedesktop.org/fontconfig/fontconfig/-/issues/472
https://github.com/lovell/sharp/issues/4429

Upstream-Status: Inappropriate [disable feature]
This is specific to libvips' prebuilt binaries and can be removed
once downstream dependencies of fontconfig handle this properly.

diff --git a/src/fcobjs.c b/src/fcobjs.c
index 1111111..2222222 100644
--- a/src/fcobjs.c
+++ b/src/fcobjs.c
@@ -82,17 +82,11 @@ static FcObjectType *
 _FcObjectLookupOtherTypeByName (const char *str, FcObject *id)
 {
     struct FcObjectOtherTypeInfo *ots, *ot;
-    static FcBool                 warn = FcFalse;
 
 retry:
     ots = fc_atomic_ptr_get (&other_types);
-    if (obj_ref.count < 1) {
-	if (!warn) {
-	    fprintf (stderr, "Fontconfig warning: using without calling FcInit()\n");
-	    warn = FcTrue;
-	}
+    if (obj_ref.count < 1)
 	FcObjectInit();
-    }
 
     for (ot = ots; ot; ot = ot->next)
 	if (0 == strcmp (ot->object.object, str))

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Mon, 15 Sep 2025 13:56:39 +0200
Subject: [PATCH 3/3] meson: Don't dllexport when built as static library on Win32

This prevents a statically built Fontconfig from exporting its
symbols into the ABI of any dynamic library it is linked into.

Upstream-Status: Backport [https://gitlab.freedesktop.org/fontconfig/fontconfig/-/commit/25092502af052bf572184401ff0c81f9968b3e53]

diff --git a/src/meson.build b/src/meson.build
index 1111111..2222222 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -1,7 +1,7 @@
 
 win_export_args = []
 # Define FcPublic appropriately for exports on windows
-if host_machine.system() == 'windows'
+if host_machine.system() == 'windows' and get_option('default_library') != 'static'
   win_export_args += '-DFcPublic=__declspec(dllexport)'
   win_export_args += '-DDLL_EXPORT'
 endif
